modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableRowSkinBase.java

```diff
@@ -349,12 +349,13 @@ protected ObjectProperty<Node> graphicProperty() {
                 height = h;
             }
 
-            width = tableCell.prefWidth(height);
-
             if (isVisible) {
                 if (fixedCellSizeEnabled && tableCell.getParent() == null) {
                     getChildren().add(tableCell);
                 }
+                // Note: We have to determine the pref width here because the add operation above may trigger the skin
+                // creation first, which is what makes it possible to get a correct value here in the first place.
+                width = tableCell.prefWidth(height);
```

Comment:
Alternatively, you could have just kept the first call to `tableCell::prefWidth` as it was, and add a second one only inside the above if expression, right after the tableCell is added to the tableRow. 
I take that, only for such case, there would be two calls instead of one, but it seems to be somehow a cleaner patch and explanation

Reply:
True and I also thought about it, but I'm still in favor of calling it only as much as needed